sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareFileOpenWriteOptions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Storage.Files.Shares.Models
+{
+    /// <summary>
+    /// Optional parameters for File Open Write.
+    /// </summary>
+    public class ShareFileOpenWriteOptions
+    {
+        /// <summary>
+        /// Whether the upload should overwrite any existing blobs.
+        /// The default value is false.
+        /// </summary>
+        public bool Overwrite { get; set; }
+
+        /// <summary>
+        /// The size of the buffer to use.  Default is 1 MB,
+        /// max is 4 MB.
+        /// </summary>
+        public long BufferSize { get; set; }
+
+        /// <summary>
+        /// Required if Overwrite is set to true.
+        /// Specifies the size of the new Page Blob.
+        /// </summary>
+        public long MaxSize { get; set; }
```

Comment:
Page blob and datalake just used `Size`. Is this more common terminology in file shares? Are files actually not always the size they're given?

Reply:
SMB Files have a max size, which is specified when you [create them](https://docs.microsoft.com/en-us/rest/api/storageservices/create-file#request-headers).